tools/lsp/server_loop.rs

```diff
@@ -570,3 +580,50 @@ fn get_code_lenses(
         .collect::<Vec<_>>();
     Some(r)
 }
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+
+    pub fn empty_document_cache(style: &str) -> DocumentCache {
+        let mut config =
+            CompilerConfiguration::new(i_slint_compiler::generator::OutputFormat::Interpreter);
+        config.style = Some(style.to_string());
+        DocumentCache::new(config)
+    }
+
+    pub fn loaded_document_cache(
+        style: &str,
+        content: String,
+        test: impl Fn(DocumentCache, &lsp_types::Url, HashMap<Url, Vec<lsp_types::Diagnostic>>) -> (),
```

Comment:
Just curious (no criticism): Is there a particular reason for using a closure and wrapping the test at the call site instead of returning the cache, url and diagnostics?

Reply:
I did not want to handle the async setup in all tests.

Reply:
Where in the tests do you need async? I mean, AFAIK the only thing in the LSP that is async is the document cache population, because of our callback API. So I don't think you need async for anything else and it should be safe to just return the populated cache, diagnostics, etc.